muse/muse.py

```diff
@@ -70,6 +71,11 @@ def connect(self, interface=None, backend='auto'):
                 self.address = address
         self.device = self.adapter.connect(self.address)
 
+        self.ask_reset()
```

Comment:
Instead of having it reset everytime, just wrap this code block in a try catch.

For example: 
```python
        try:
            self.device = self.adapter.connect(self.address)
            # subscribes to EEG stream
            if self.enable_eeg:
                self._subscribe_eeg()

            if self.enable_control:
                self._subscribe_control()

            if self.enable_telemetry:
                self._subscribe_telemetry()

            if self.enable_acc:
                self._subscribe_acc()

            if self.enable_gyro:
                self._subscribe_gyro()

        except pygatt.exceptions.BLEError as error:
        if("characteristic" in error.__str__):
            self.ask_reset()
            sleep(2)
            self.device = self.adapter.connect(self.address)
            self.select_preset(preset=21)

            # subscribes to EEG stream
            if self.enable_eeg:
                self._subscribe_eeg()

            if self.enable_control:
                self._subscribe_control()

            if self.enable_telemetry:
                self._subscribe_telemetry()

            if self.enable_acc:
                self._subscribe_acc()

            if self.enable_gyro:
                self._subscribe_gyro()

Reply:
I agree doing an ask_device_info callback would be the best way to do that. However, @sylvchev's right in that it's kind of tricky to do that kind of async/await logic in Python

Reply:
Unfortunately, I have currently no access to a Muse band. I could not test your solution but it seems a better solution than to reset everytime. 
@alexandrebarachant, @hubertjb, if you agree, I'll update my PR with the code proposed by @jdpigeon.